### PR TITLE
adding a new flag for github-auth-token

### PR DESCRIPTION
**Pull request by acornett21**
- Fixes: #9
- adding new flag to accept a github-auth-token.
- using this flag to determine if we should create a github client with an auth token or not.

**NOTE: This was created off another branch to have the necessary lib to call the WithAuthToken method. This will be removed on rebase, after the other PR is merged***

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -4,11 +4,12 @@ Copyright © 2022 NAME HERE <EMAIL ADDRESS>
 package cmd
 
 import (
-	configflagutil "k8s.io/test-infra/prow/flagutil/config"
 	"log"
 	"os"
 	"reflect"
 
+	configflagutil "k8s.io/test-infra/prow/flagutil/config"
+
```
Comment (bcrochet) [inline cmd/root.go:12]: No need for the empty line here.

**Review by bcrochet — COMMENTED** [review]
Just a comment about an import section. Will revisit when prior changes have landed.

File: cmd/root.go (hunk 1 above)
Comment (acornett21) [reply cmd/root.go:12]: good catch, will check what `go fmt` did on this.

**Comment by mrhillsman** [comment]
No need to use a flag the environment variable approach is easy enough. https://github.com/google/go-github/blob/master/test/fields/fields.go#L45

**Comment by acornett21** [comment]
> No need to use a flag the environment variable approach is easy enough.

IMO, it makes for a bad user experience to expose some things as flags and others as env's. This is also confusing for the user since ENV's would be hidden from a 'help' cmd. Adding as a flag, we also get the ENV for free, so it makes sense to keep things consistent and add a flag.

**Comment by mrhillsman** [comment]
> > No need to use a flag the environment variable approach is easy enough.
> 
> IMO, it makes for a bad user experience to expose some things as flags and others as env's. This is also confusing for the user since ENV's would be hidden from a 'help' cmd. Adding as a flag, we also get the ENV for free, so it makes sense to keep things consistent and add a flag.

What I am saying is preflight-trigger itself does not care about the Github token it is not even required or necessary for preflight-trigger. Documenting the Github token env var takes as much work as documenting the flag. Rather than touching all the other files with the code you currently have the env var in just the github.go file and updating help to note a token env var is touching less things.

**Comment by Allda** [comment]
> > > No need to use a flag the environment variable approach is easy enough.
> > 
> > 
> > IMO, it makes for a bad user experience to expose some things as flags and others as env's. This is also confusing for the user since ENV's would be hidden from a 'help' cmd. Adding as a flag, we also get the ENV for free, so it makes sense to keep things consistent and add a flag.
> 
> What I am saying is preflight-trigger itself does not care about the Github token it is not even required or necessary for preflight-trigger itself to work. Documenting the Github token env var takes as much work as documenting the flag. Rather than touching all the other files with the code you currently have the env var in just the github.go file and updating help to note a token env var is touching less things. We are only addressing the case brought by the community operators folks which has not been 100% verified that preflight-trigger is the root cause.

I am also leaning towards using env variables for the token. From my experience, it is really easy to leak a token when cli tools require it as a cli argument. When using env variable it is usually a matter of updating the config in the ocp pod/tekton task with no need to update the command itself. Also, the config is optional (is really used just to bump up the API limit).

**Comment by bcrochet** [comment]
> > > No need to use a flag the environment variable approach is easy enough.
> > 
> > 
> > IMO, it makes for a bad user experience to expose some things as flags and others as env's. This is also confusing for the user since ENV's would be hidden from a 'help' cmd. Adding as a flag, we also get the ENV for free, so it makes sense to keep things consistent and add a flag.
> 
> What I am saying is preflight-trigger itself does not care about the Github token it is not even required or necessary for preflight-trigger itself to work. Documenting the Github token env var takes as much work as documenting the flag. Rather than touching all the other files with the code you currently have the env var in just the github.go file and updating help to note a token env var is touching less things. We are only addressing the case brought by the community operators folks which has not been 100% verified that preflight-trigger is the root cause.

While I do agree that the flag is problematic, (as @Allda mentions) I do think passing the client in to the function instead of a package level variable is a good idea.

**Comment by acornett21** [comment]
I think I addressed the concern with the latest push, though I'd like PR #13 to merge first so this PR has proper build checks.

File: cmd/job.go [hunk 2]
```diff
@@ -81,7 +83,14 @@ func jobPreRun(cmd *cobra.Command, args []string) {
 
 	validateJobFlags()
 
-	config, err := GetGitHubFile("openshift", "release", "core-services/prow/02_config/_config.yaml")
+	// checking to see if an auth token was set as an ENV, if so create the gh client with the token
+	githubAuthToken := os.Getenv("GITHUB_AUTH_TOKEN")
+	githubClient := github.NewClient(nil)
+	if githubAuthToken != "" {
+		githubClient = githubClient.WithAuthToken(githubAuthToken)
+	}
```
Comment (bcrochet) [inline cmd/job.go:91]:
```suggestion
	// checking to see if an auth token was set as an ENV, if so create the gh client with the token
	githubClient := github.NewClient(nil)
	if githubAuthToken, found := os.LookupEnv("GITHUB_AUTH_TOKEN"); found {
		githubClient = githubClient.WithAuthToken(githubAuthToken)
	}
```

**Review by bcrochet — CHANGES_REQUESTED** [review]
Slight simplification.

**Review by mrhillsman — CHANGES_REQUESTED** [review]
Guys we are making this too complex. Just do the following to github.go:

import os

func GetGitHubFile(owner, repo, path string) (string, error) {
	token, found := os.LookupEnv("GITHUB_TOKEN"); if found && token != "" {
		client = github.NewClient(nil).WithAuthToken(token)
	} else {
		client = github.NewClient(nil)
	}

**Comment by bcrochet** [comment]
> Guys I think we are making this too complex. Just do the following to github.go if not wanting a global var. Again I don't think any other files need to be changed considering the usage:
> 
> ```go
> import os
> 
> func GetGitHubFile(owner, repo, path string) (string, error) {
> 	token, found := os.LookupEnv("GITHUB_TOKEN"); if found && token != "" {
> 		client = github.NewClient(nil).WithAuthToken(token)
> 	} else {
> 		client = github.NewClient(nil)
> 	}
> ```

I can make it more complicated if you want... :)

I think passing in the client is ideal, but that could be a future refactoring.

**Comment by mrhillsman** [comment]
> > Guys I think we are making this too complex. Just do the following to github.go if not wanting a global var. Again I don't think any other files need to be changed considering the usage:
> > ```go
> > import os
> > 
> > func GetGitHubFile(owner, repo, path string) (string, error) {
> > 	token, found := os.LookupEnv("GITHUB_TOKEN"); if found && token != "" {
> > 		client = github.NewClient(nil).WithAuthToken(token)
> > 	} else {
> > 		client = github.NewClient(nil)
> > 	}
> > ```
> 
> I can make it more complicated if you want... :)
> 
> I think passing in the client is ideal, but that could be a future refactoring.

:)

**Review by mrhillsman — APPROVED** [review]
/lgtm

**Review by bcrochet — APPROVED** [review]
/lgtm

**Pull request merged**